src/main/java/com/netgrif/application/engine/elastic/service/ElasticTaskQueueManager.java

```diff
@@ -0,0 +1,166 @@
+            switch (task.getTypeJob()) {
+                case INDEX:
+                    return indexTaskWorker(task.getTask());
+                case REMOVE:
+                    return removeTaskWorker(task.getTask());
+                default:
+                    log.warn("Unknown task type for task: {}", task.getTask().getTaskId());
+                    return null;
+            }
+        } catch (Exception e) {
+            log.error("Error processing task {}: {}", task.getTask().getTaskId(), e.getMessage(), e);
+            throw e;
+        } finally {
+            activeTasks.remove(task.getTask().getTaskId());
+            scheduleNextTask(task.getTask().getTaskId());
+        }
+    }
+
+
+    private void scheduleNextTask(String taskId) {
+        BlockingQueue<Runnable> queue = taskQueues.get(taskId);
+        if (queue != null) {
+            Runnable nextTask = queue.poll();
+            if (nextTask != null) {
+                log.debug("Submitting next task for ID: {} to executor", taskId);
+                elasticTaskExecutor.submit(nextTask);
+            } else {
+                activeTasks.remove(taskId);
+                taskQueues.remove(taskId);
+            }
+        }
+    }
+
+
+    @PreDestroy
+    public void destroy() throws InterruptedException {
+        log.info("Shutting down ElasticTaskQueueManager");
+        elasticTaskExecutor.shutdown();
+    }
+
+
+    private ElasticTask indexTaskWorker(ElasticTask task) {
+        log.debug("Indexing task [{}] in thread [{}]", task.getTaskId(), Thread.currentThread().getName());
+        ElasticTask elasticTask = null;
+        try {
+            elasticTask = repository.findByStringId(task.getStringId());
+            if (elasticTask == null) {
+                elasticTask = repository.save(task);
+            } else {
+                elasticTask.update(task);
+                elasticTask = repository.save(elasticTask);
+            }
+            log.debug("[{}]: Task \"{}\" [{}] indexed", task.getCaseId(), task.getTitle(), task.getStringId());
+        } catch (InvalidDataAccessApiUsageException e) {
+            log.debug("[{}]: Task \"{}\" has duplicates, will be reindexed", task.getCaseId(), task.getTitle());
+            repository.deleteAllByStringId(task.getStringId());
+            repository.save(task);
+            log.debug("[{}]: Task \"{}\" indexed", task.getCaseId(), task.getTitle());
+        } catch (RuntimeException e) {
+            log.error("Elastic executor was killed before finish: {}", e.getMessage());
+        }
+        return elasticTask;
+    }
+
+    private ElasticTask removeTaskWorker(ElasticTask task) {
+        log.debug("Remove task [{}] in thread [{}]", task.getTaskId(), Thread.currentThread().getName());
+        try {
+            log.debug("[{}]: Task \"{}\" [{}] removed", task.getCaseId(), task.getTitle(), task.getStringId());
+            return repository.deleteAllByTaskId(task.getTaskId());
```

Comment:
in the `indexTaskWorker` method deletion is done by `stringId` here by `taskId`, why?

Reply:
Done